src/indicator/algorithm/calculator/stochasticOscillator.js

```diff
@@ -0,0 +1,68 @@
+import d3 from 'd3';
+import _slidingWindow from './slidingWindow';
+import {rebind} from '../../../util/rebind';
+
+export default function() {
+
+    var closeValue = function(d, i) { return d.close; },
+        highValue = function(d, i) { return d.high; },
+        lowValue = function(d, i) { return d.low; };
+
+    var kWindow = _slidingWindow()
+        .windowSize(5)
+        .accumulator(function(values) {
+            var maxHigh = d3.max(values, highValue);
+            var minLow = d3.min(values, lowValue);
+            return 100 * (closeValue(values[values.length - 1]) - minLow) / (maxHigh - minLow);
+        });
+
+    var dWindow = _slidingWindow()
+        .windowSize(3)
+        .accumulator(function(values) {
+            if (values[0] === undefined) {
+                return undefined;
+            }
+            return d3.mean(values);
+        });
+
+    var stochastic = function(data) {
+        var kValues = kWindow(data);
+        var dValues = dWindow(kValues);
+        return kValues.map(function(k, i) {
```

Comment:
Ah, it would appear that the d3 zip behaviour is different from underscores! (Which basically does what we want here)

The main advantage we get from zip (or similar) is a clearer statement of intent.